### DISCUSSION TITLE
How to add architecture into shortcuts?

### DISCUSSION THREAD

**Discussion by Xelbayria** [ScoopInstaller/Scoop#5476] [Help and Questions] [answered]
Hello, 

to quote from the documentation: 
```
architecture: If the app has 32- and 64-bit versions, architecture can be used to wrap the differences 
([example](https://github.com/ScoopInstaller/Main/blob/master/bucket/7zip.json)).

    32bit|64bit: contains architecture-specific instructions (bin, checkver, extract_dir, hash, installer, 
pre_install, post_install, shortcuts, uninstaller, url, and msi [msi is deprecated]).
```

Does this mean I can do like this: 
```json
"architecture": {
    "64bit": {
		"shortcuts": [
			[ 
				"filename.exe",
				"name"
			]
		]
	},
	"32bit": {
		"shortcuts": [
			[ 
				"filename.exe",
				"name"
			]
		]
	}
} 
```

or is it the other way around? 

Edit: 
There is no 64bits or 32bits for the URL or HASH.  but there are 32bits & 64bits for executable files in "Bin".  There is also a normal executable files in the root of the folder

**Comment by HUMORCE** [comment, marked as answer]
A:
Shortcut points to the corresponding executable file.

```
    "url": "https://example.org/app.zip",
    "hash": "<value>",
    "architecture": {
        "64bit": {
            "shortcuts": [
                [
                    "app-64bit.exe",
                    "AppName"
                ]
            ]
        },
        "32bit": {
            "shortcuts": [
                [
                    "app-32bit.exe",
                    "AppName"
                ]
            ]
        }
    },
```

B:
Rename the corresponding executable file to a fixed name.

```
    "url": "https://example.org/app.zip",
    "hash": "<value>",
    "architecture": {
        "64bit": {
            "pre_install": "Rename-Item \"$dir\\app_64bit.exe\" 'app.exe'; Remove-Item \"$dir\\app_32bit.exe\"",
        },
        "32bit": {
            "pre_install": "Rename-Item \"$dir\\app_32bit.exe\" 'app.exe'; Remove-Item \"$dir\\app_64bit.exe\"",
        }
    },
    "shortcuts": [
        [
            "app.exe",
            "AppName"
        ]
    ],
```

C:
Create both 64/32 bit shortcuts for 64bit installations.

```
    "url": "https://example.org/app.zip",
    "hash": "<value>",
    "architecture": {
        "64bit": {
            "shortcuts": [
                [
                    "app-64bit.exe",
                    "AppName"
                ],
                [
                    "app-32bit.exe",
                    "AppName (32bit)"
                ],
            ]
        },
        "32bit": {
            "shortcuts": [
                [
                    "app-32bit.exe",
                    "AppName"
                ]
            ]
        }
    },
```

**Comment by Xelbayria** [comment]
Just to check the order is correct: 
        -   `version`
        -   `description`
        -   `homepage`
        -   `license`
        -   `notes`
        -   `depends`
        -   `suggest`
        -   `url`
        -   `hash`
        -  `architecture` <- Added
        -   `extract_dir`
        -   `extract_to`
        -   `pre_install`
        -   `installer`
        -   `post_install`
        -   `env_add_path`
        -   `env_set`
        -   `bin`
        -   `shortcuts`
        -   `persist`
        -   `uninstaller`
        -   `checkver`
        -   `autoupdate`

**Comment by HUMORCE** [comment]
It takes the place of `shortcuts` when there are only `shortcuts` in the architecture. this is not mandatory, but most manifests follow this rule.

```
    "url": "https://example.org/app.zip",
    "hash": "<value>",
    "extract_dir": "foo",
    "pre_install": "info 'Installing <App>'",
    "architecture": {
        "64bit": {
            "bin": [
                [
                    "app-64bit.exe",
                    "app"
                ]
            ],
            "shortcuts": [
                [
                    "app-64bit.exe",
                    "AppName"
                ],
                [
                    "app-32bit.exe",
                    "AppName (32bit)"
                ],
            ]
        },
        "32bit": {
            "bin": [
                [
                    "app-32bit.exe",
                    "app"
                ]
            ],
            "shortcuts": [
                [
                    "app-32bit.exe",
                    "AppName"
                ]
            ]
        }
    },
    "persist": "configs",
```

**Reply by Xelbayria** [reply]
> It takes the place of `shortcuts` when there are only `shortcuts` in the architecture. this is not mandatory, but most manifests follow this rule.
> 
> ```
>     "url": "https://example.org/app.zip",
>     "hash": "<value>",
>     "extract_dir": "foo",
>     "pre_install": "info 'Installing <App>'",
>     "architecture": {
>         "64bit": {
>             "bin": [
>                 [
>                     "app-64bit.exe",
>                     "app"
>                 ]
>             ],
>             "shortcuts": [
>                 [
>                     "app-64bit.exe",
>                     "AppName"
>                 ],
>                 [
>                     "app-32bit.exe",
>                     "AppName (32bit)"
>                 ],
>             ]
>         },
>         "32bit": {
>             "bin": [
>                 [
>                     "app-32bit.exe",
>                     "app"
>                 ]
>             ],
>             "shortcuts": [
>                 [
>                     "app-32bit.exe",
>                     "AppName"
>                 ]
>             ]
>         }
>     },
>     "persist": "configs",
> ```

Yes I am aware of this is not mandatory. The Manifest rule is missing a few keys like "Architecture" and also the manifest.json is also missing "shortcuts", too.   I am talking about the template to use to create a new manifest for the software.  I need to know for sure, so i can update my note in my note-taking app.

**Reply by HUMORCE** [reply]
Wiki or Template repo:
https://github.com/ScoopInstaller/Scoop/wiki/App-Manifests
https://github.com/ScoopInstaller/BucketTemplate/blob/master/bucket/app-name.json.template

**Reply by Xelbayria** [reply]
> Wiki or Template repo: https://github.com/ScoopInstaller/Scoop/wiki/App-Manifests https://github.com/ScoopInstaller/BucketTemplate/blob/master/bucket/app-name.json.template

I already have these. I want to update the information. that's all.   this is what I am referring to: [For Scoop Bucket](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md)
